telegram/message.py

```diff
@@ -338,10 +338,14 @@ def chat_id(self):
 
     @property
     def link(self):
-        """:obj:`str`: Convenience property. If the chat of the message is a supergroup or a
-        channel and has a :attr:`Chat.username`, returns a t.me link of the message."""
-        if self.chat.type in (Chat.SUPERGROUP, Chat.CHANNEL) and self.chat.username:
-            return "https://t.me/{}/{}".format(self.chat.username, self.message_id)
+        """:obj:`str`: Convenience property. If the chat of the message is not
+        a private chat, returns a t.me link of the message."""
+        if self.chat.type is not Chat.PRIVATE:
```

Comment:
At least on my machine, for some reason only `if self.chat.type != Chat.PRIVATE` will work. With `is not` it will also return a link in private chats.